gravitee-plugin-repository/src/main/java/io/gravitee/plugin/repository/internal/RepositoryTypeReader.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.plugin.repository.internal;
+
+import io.gravitee.platform.repository.api.Scope;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.env.Environment;
+
+@RequiredArgsConstructor
+class RepositoryTypeReader {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RepositoryTypeReader.class);
+
+    String getRepositoryType(Environment environment, Scope scope) {
+        String repositoryType = environment.getProperty("repositories." + scope.getName() + ".type");
+        if (repositoryType != null) {
+            return repositoryType;
+        } else {
+            repositoryType = environment.getProperty(scope.getName() + ".type");
+            if (repositoryType != null) {
+                LOGGER.warn(
+                    "Repository of scope {} is loaded from old structure. Please move repositories settings under 'repositories' section.",
```

Comment:
"old structure" is not clear.
Maybe this message would be clearer:
"'<scope>.type' configuration parameter is discouraged, please use 'repositories.<scope>.type'"
where <scope> is the scope value.

I'm wondering if we shouldn't only log this warning when the `node.application()` value starts with "gio-am"

Reply:
How about this?
> The repository of scope MANAGEMENT is loaded from discouraged section 'management.type'. Please use 'repositories.management.type'.

Reply:
And also I don't really understand what node.application() is. What component do you mean by that?

Reply:
My understanding is, the APIM will update it when 4.5.0 is released. Till then, they can use current version?

Reply:
Agreed to keep warn message regardless the node.applicaiton() value